utils/extremum_tracking.hh

```diff
@@ -10,41 +10,33 @@
 
 #include <functional>
 #include <limits>
+#include <optional>
 
 namespace detail {
 
 template<typename T, typename Comparator>
 class extremum_tracker {
     T _default_value;
-    bool _is_set = false;
-    T _value;
+    std::optional<T> _value;
 public:
-    explicit extremum_tracker(T default_value) {
-        _default_value = default_value;
-    }
+    explicit extremum_tracker(const T& default_value)
```

Comment:
Better take value type and move it into `_default_value`. Might be a moot point for the types used here (integers).

Reply:
I see this is done later, resolving comment.